### PR TITLE
Added Fastlane workflows

### PR DESCRIPTION
**Pull request by batesyuk**

### CODE REVIEW

File: Fastlane/Configuration.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'yaml'
+
+class Configuration
+
+  attr_reader :build_number
+  attr_reader :app_scheme, :app_output_directory, :app_display_name
+  attr_reader :scan_devices
```
Comment (1pretz1) [inline Fastlane/Configuration.rb:7]: We don't need any of the readers above as the methods already exist and are publicly accessible. We should however have them for `@lane_name` & `@yaml` so we don't need to call instance variables directly from within the class (+ encapsulate within private to limit access outside), e.g:

```
...
def initialize(lane_name, yaml)
    @lane_name = lane_name
    @yaml = YAML.load_file(yaml, aliases: true)[lane]
  end
...
private

attr_reader :lane_name, :yaml

def blah
  lane_name
end
```

File: Fastlane/Configuration.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  private
+  def setting(prefix, key)
+    @yaml[prefix.to_s][key.to_s]
+  end
+
+  def lane
+    @lane ||= @lane_name.split(' ').last
+  end
+
```
Comment (1pretz1) [inline Fastlane/Configuration.rb:42]: Minor: remove space above

File: Fastlane/Configuration.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  def app_display_name
+    setting(:app, :display_name)
+  end
+
+  def scan_devices
+    setting(:scan, :devices)
+  end
+
+  private
```
Comment (1pretz1) [inline Fastlane/Configuration.rb:34]: Minor: Add space below

File: .github/workflows/pull_request.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          lfs: 'true'
+          fetch-depth: 0
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
```
Comment (1pretz1) [inline .github/workflows/pull_request.yaml:25]: Nice! Even though macOS builds with system Ruby, it's a good idea to have versioning 👍

File: .github/workflows/pull_request.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+name: iOS Pull Request
+
+on:
+  pull_request:
```
Comment (1pretz1) [inline .github/workflows/pull_request.yaml:4]: Do we also want CI to be triggered when a branch push occurs? Sometimes useful when testing changes vs having to open a PR

File: Fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    gym(
+      scheme: configuration.app_scheme,
+      output_directory: configuration.app_output_directory,
+      clean: options[:clean].nil? ? true : options[:clean],
+      skip_profile_detection: true
+      )
+  end
+
+
+end
```
Comment (1pretz1) [inline Fastlane/Fastfile:37]: Minor: 2 extra spaces above 37

File: Fastlane/Fastfile [hunk 7]
```diff
@@ -0,0 +1,44 @@
+require_relative 'Configuration'
+
+default_platform :ios
+
+platform :ios do
```
Comment (1pretz1) [inline Fastlane/Fastfile:5]: Minor: remove space below

File: Fastlane/Fastfile [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  lane :unit do
+    unit_test
+  end
+
+  desc "Builds alpha configuration"
+  lane :alpha do
+    build
+  end
+
+  private_lane :unit_test do
```
Comment (1pretz1) [inline Fastlane/Fastfile:17]: End parentheses brought inline with method call. Same comment for other two calls:
 
- https://github.com/alphagov/govuk-mobile-ios-app/pull/2/files#diff-fb2c7d64b89b76b2a315e5a7365ee1f5f55938bec1a2a766bc2d1c623bda9cb4R28)
- https://github.com/alphagov/govuk-mobile-ios-app/pull/2/files#diff-fb2c7d64b89b76b2a315e5a7365ee1f5f55938bec1a2a766bc2d1c623bda9cb4R43

```
    scan(
      scheme: configuration.app_scheme,
      devices: configuration.scan_devices,
      clean: false,
      prelaunch_simulator: true,
      derived_data_path: './build_for_unit_testing'
    )
```

File: Fastlane/Fastfile [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      prelaunch_simulator: true,
+      derived_data_path: './build_for_unit_testing'
+      )
+  end
+
+  private_lane :build do |options|
+    gym(
+      scheme: configuration.app_scheme,
+      output_directory: configuration.app_output_directory,
+      clean: options[:clean].nil? ? true : options[:clean],
```
Comment (1pretz1) [inline Fastlane/Fastfile:31]: We don't need `.nil?` here for the ternary operator as Ruby will return false if options[:clean] is `nil`, also could probs simplify (if options[:clean] is anything but `nil` or `false` then return options[:clean], else return `true`)
```
options[:clean] || true
```

**Review by 1pretz1 — COMMENTED** [review]
Nice work! Left some minor comments / questions. I've also embodied a linter (😄) so the most obvious step would be to add one but may not be worth it due to the limited Ruby code we're adding. 

Could also just install the `rubocop` gem, and run `rubocop -A` locally which will autofix / suggest fixes

File: Fastlane/Configuration.rb (hunk 1 above)
Comment (batesyuk) [reply Fastlane/Configuration.rb:7]: lol, so I have all the ones I don't need and none of the ones I do need? Classic!

File: .github/workflows/pull_request.yaml (hunk 5 above)
Comment (batesyuk) [reply .github/workflows/pull_request.yaml:4]: Yeah, I will make more workflows for those and one that can be triggered manually

File: Fastlane/Configuration.rb [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  attr_reader :build_number
+  attr_reader :app_scheme, :app_output_directory, :app_display_name
+  attr_reader :scan_devices
+
+  def initialize(lane_name, yaml)
+    @lane_name = lane_name
+    @yaml = YAML.load_file(yaml, aliases: true)[lane]
+  end
+
+  def build_number
```
[the remark this answers is not in the document]
Comment (batesyuk) [reply Fastlane/Configuration.rb:12]: display name will be used for alpha and release,
lane is used
build_number is used for release builds

**Review by 1pretz1 — APPROVED** [review]
Great work Tom, workflows look great! Ideally commits will be tidied up before merge 😄

**Pull request merged**